src/sentry/middleware/integrations/parsers/msteams.py

```diff
@@ -49,7 +51,12 @@ def get_response(self) -> HttpResponseBase:
         if not self.can_infer_integration(data=self.request_data):
             return self.get_response_from_control_silo()
 
-        regions = self.get_regions_from_organizations()
+        regions: Sequence[Region] = []
+        try:
+            regions = self.get_regions_from_organizations()
+        except (Integration.DoesNotExist, OrganizationIntegration.DoesNotExist):
+            pass
```

Comment:
This may be safe to call return `self.get_default_missing_integration_response()` here, since it looks like the fallback to control silo is only if regions aren't found, and like you mentioned, might result in a cross-silo query.

@dashed might know the specifics since these extra scopes were part of his debugging.